rts/Rendering/Env/Particles/ProjectileDrawer.cpp

```diff
@@ -42,11 +42,11 @@ CONFIG(int, SoftParticles).defaultValue(1).safemodeValue(0).description("Soften
 
 
 static bool CProjectileDrawOrderSortingPredicate(const CProjectile* p1, const CProjectile* p2) noexcept {
-	return std::tie(p2->drawOrder, p1->GetSortDist(), p1) > std::tie(p1->drawOrder, p2->GetSortDist(), p2);
+	return std::make_tuple(p2->drawOrder, p1->GetSortDist(), p1) > std::make_tuple(p1->drawOrder, p2->GetSortDist(), p2);
```

Comment:
There's also `std::forward_as_tuple` which seems closer to `std::tie` in that it still uses references where possible, but all 3 of these are simple types that don't get modified so a copy should still be better.

Reply:
Here all arguments are trivial types, I doubt we will gain anything by making them references.